go/common/witness/proof.go

```diff
@@ -23,13 +23,30 @@ import (
 // the full state to extract information.
 type Proof interface {
 
+	// Extract extracts a sub-proof for a given account and selected storage locations from this proof.
+	// It returns a copy that contains only the data necessary for proving the given address and storage keys.
+	// The resulting proof covers proofs for the intersection of the requested properties (account information and slots)
+	// and the properties covered by this proof. The second return parameter indicates whether everything that
+	// was requested could be covered. If so it is set to true, otherwise it is set to false.
+	Extract(root common.Hash, address common.Address, keys ...common.Key) (Proof, bool)
+
 	// IsValid checks that this proof is self-consistent. If the result is true, the proof can be used
 	// for extracting verified information. If false, the proof is corrupted and should be discarded.
 	IsValid() bool
 
 	// GetElements returns serialised elements of the witness proof.
 	GetElements() []string
 
+	// GetStorageElements returns serialised elements of the witness proof for a given account
+	// and selected storage locations from this proof.
+	// The resulting proof contains only a proof for the storage part of the account.
```

Comment:
nit: the documentation states that the first result value is a proof, but it is a `[]string` instead. Should the result be replaced by a proof?

Reply:
thx, fixed doc. I did not want to return the Proof interface - since it would contain only storage nodes, all getter of the Proof interface would be useless because we do not support proving storage only.